README.md
# WooCommerce Stripe fee in Report
[![Donate](https://img.shields.io/badge/Donate-PayPal-green.svg)](https://www.paypal.me/rvola)

Shows Stripe fees in WooCommerce charts. Calculates a correct net total.

![banner](/.github/banner.jpg)

## Description

**WooCommerce Stripe fee in Report** is a small addon that will allow you to quickly visualize in the WooCommerce graph, the state of your comissions Stripe !!
It's simple and it works.

## Installation

This section describes how to install the plugin and get it working.

1. Upload the plugin files to the `/wp-content/plugins/woo-stripe-fee-in-report` directory, or install the plugin through the WordPress plugins screen directly.
2. Activate the plugin through the 'Plugins' screen in WordPress.
3. Enjoy ! 🤑

## Requirement

* PHP minimal: **7.0**
* WordPress minimal: **4.4**
* WooCommerce minimal : **3.1**
* WooCommerce Gateway Stripe minimal : **3.0**

## Links

* [**Changelog**](https://github.com/rvola/woo-stripe-fee-in-report/blob/master/CHANGELOG.md)
* [**Download on WordPress**](https://wordpress.org/plugins/woo-stripe-fee-in-report/)




CHANGELOG.md
# Changelog

## [1.2.6](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.6) - 2020-08-24
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.2.5...1.2.6)

* ✔︎ Compatibility WOO 4.4
* ✔︎ Compatibility WP 5.5

## [1.2.5](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.5) - 2020-03-09
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.2.4...1.2.5)

* ✔︎ Compatibility WOO 4.0
* ✔︎ Compatibility WP 5.4

## [1.2.4](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.4) - 2020-01-22
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.2.3...1.2.4)

* ✔︎ Compatibility WOO 3.9

## [1.2.3](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.3) - 2019-11-15
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.2.2...1.2.3)

* FIX / Incorrect call folder

## [1.2.2](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.2) - 2019-10-23
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.2.1...1.2.2)

* FIX / Load languages files
* escape i18n
* FIX / Series incorrect for legend + escape i18n
* ✔︎ Compatibility WOO 3.8
* ✔︎ Compatibility WP 5.3

## [1.2.1](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.1) - 2019-04-03
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.2.0...1.2.1)

* ✔︎ Compatibility WP 5.2
* ✔︎ Compatibility WOO 3.6

## [1.2.0](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.2.0) - 2019-03-08
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.1.4...1.2.0)

* Add data to CSV export
* ✔ Check compatbility WP 5.1
* FIX / url GitHub

## [1.1.4](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.1.4) - 2018-10-22
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.1.3...1.1.4)

* End support PHP 5.6
* ✔︎ Check compatbility WP 5.0

## [1.1.3](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.1.3) - 2018-10-02
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.1.2...1.1.3)

* ✔︎ Compatibility WooCommerce 3.5

## [1.1.2](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.1.2) - 2018-06-04
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.1.1...1.1.2)

* Fix - Error in calculating net total #2

## [1.1.1](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.1.1) - 2018-05-23
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.1.0...1.1.1)

* ✔︎ Compatibility WooCommerce 3.4

## [1.1.0](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.1.0) - 2018-03-20
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.0.2...1.1.0)

* Updated meta_key name query (Stripe Gateway 4.1)
* Use constant for regex query

## [1.0.2](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.0.2) - 2018-03-01
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.0.1...1.0.2)

* NEW Define WOOSFR_FILE
* Refactor instance

## [1.0.1](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.0.1) - 2018-02-25
[Full Changelog](https://github.com/rvola/woo-stripe-fee-in-report/compare/1.0.0...1.0.1)

* Change yaxis for shipping
* Change yaxis for Stripe fees
* Rename path (plugin submission)

## [1.0.0](https://github.com/rvola/woo-stripe-fee-in-report/tree/1.0.0) - 2018-02-24

* Launch


includes/class-wp.php
<?php
/**
 * Initialize the plugin
 *
 * @package RVOLA\WOO\SFR
 **/

namespace RVOLA\WOO\SFR;

if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

/**
 * Class WP
 *
 * @package RVOLA\WOO\SFR
 */
class WP {

	/**
	 * Singleton
	 *
	 * @var singleton.
	 */
	private static $_singleton = null;

	/**
	 * WooSFR constructor.
	 */
	public function __construct() {
		add_action( 'init', array( $this, 'load_languages' ), 10 );
		add_filter( 'plugin_row_meta', array( $this, 'plugin_row_meta' ), 10, 2 );

		// Start Report.
		require_once dirname( __FILE__ ) . '/class-report.php';
		new Report();
	}

	/**
	 * Singleton.
	 *
	 * @return mixed
	 */
	public static function instance() {
		if ( is_null( self::$_singleton ) ) {
			self::$_singleton = new self();
		}

		return self::$_singleton;
	}

	/**
	 * Load language files
	 */
	public function load_languages() {

		load_plugin_textdomain( 'woo-stripe-fee-in-report', false, plugin_basename( dirname( WOOSFR_FILE ) ) . '/languages' );
	}

	/**
	 * Add links in the list of plugins.
	 *
	 * @param array  $plugin_meta An array of the plugin's metadata, including the version, author, author URI, and plugin URI.
	 * @param string $plugin_file Path to the plugin file, relative to the plugins directory.
	 *
	 * @return mixed
	 */
	public function plugin_row_meta( $plugin_meta, $plugin_file ) {
		if ( plugin_basename( WOOSFR_FILE ) === $plugin_file ) {
			array_push(
				$plugin_meta,
				sprintf(
					'<a href="https://www.paypal.me/rvola" target="_blank">%s</a>',
					esc_html__( 'Donate', 'woo-stripe-fee-in-report' )
				)
			);
		}

		return $plugin_meta;
	}
}


includes/class-report.php
<?php
/**
 * Class report that displays the Stripe fees.
 *
 * @package RVOLA\WOO\SFR
 **/

namespace RVOLA\WOO\SFR;

if ( ! defined( 'ABSPATH' ) ) {
	exit;
}
if ( ! defined( 'WC_ABSPATH' ) ) {
	exit;
}
require_once WC_ABSPATH . 'includes/admin/reports/class-wc-admin-report.php';

/**
 * Class Report
 *
 * @package RVOLA\WOO\SFR
 */
class Report extends \WC_Admin_Report {

	/**
	 * Color for the graph.
	 */
	const COLOR = '#572ff8';
	/**
	 * Name of the meta key for Stripe fees (old and new).
	 */
	const LEGACY_META_NAME_FEE = 'Stripe Fee';
	const META_NAME_FEE = '_stripe_fee';
	/**
	 * Name of tweak for customize query
	 */
	const TWEAK_REQ = '___woosfr___';

	/**
	 * Chart data table.
	 *
	 * @var $chart_data
	 */
	private $chart_data;
	/**
	 * Data table to build the graph.
	 *
	 * @var $report_data
	 */
	private $report_data;

	/**
	 * WooSFR constructor.
	 */
	public function __construct() {
		$this->init_variable();
		$this->init_widget_dashboard();

		add_filter( 'woocommerce_admin_report_data', array( $this, 'add_report_data' ), 10, 1 );
		add_filter( 'woocommerce_reports_get_order_report_query', array( $this, 'clean_query_get_order' ), 10, 1 );

		add_filter( 'woocommerce_admin_report_chart_data', array( $this, 'chart_add_report_data' ), 10, 1 );
		add_action( 'admin_print_footer_scripts', array( $this, 'chart_add_legend' ), 10 );
		add_action( 'admin_print_footer_scripts', array( $this, 'chart_update_legend_placeholder' ), 10 );
		add_action( 'admin_print_footer_scripts', array( $this, 'chart_draw' ), 10 );
		add_action( 'admin_print_footer_scripts', array( $this, 'chart_style' ), 10 );

	}

	/**
	 * Set the default values to start the class.
	 */
	private function init_variable() {

		$current_range = ! empty( $_GET['range'] ) ? sanitize_text_field( wp_unslash( $_GET['range'] ) ) : '7day';

		if ( ! in_array( $current_range, array( 'custom', 'year', 'last_month', 'month', '7day' ), true ) ) {
			$current_range = '7day';
		}

		$this->check_current_range_nonce( $current_range );
		$this->calculate_current_range( $current_range );
	}

	/**
	 * Adjust the "total net" value in the widget on the dashboard.
	 */
	private function init_widget_dashboard() {

		global $pagenow;

		if ( 'index.php' === $pagenow ) {
			$this->start_date = strtotime( date( 'Y-m-01', current_time( 'timestamp' ) ) );
			$this->end_date   = current_time( 'timestamp' );
		}

	}

	/**
	 * Add Stripes fees to the chart data table.
	 *
	 * @param array $report_data Data table.
	 *
	 * @return array $this->_report_data
	 */
	public function add_report_data( $report_data ) {

		// Set report_data in class.
		$this->report_data = $report_data;

		// Search Stripe fees values.
		$stripe_fees = $this->get_order_report_data( array(
			'data'         => array(
				self::TWEAK_REQ => array(
					'type'     => 'meta',
					'function' => 'SUM',
					'name'     => 'total',
				),
				'post_date'     => array(
					'type'     => 'post_data',
					'function' => '',
					'name'     => 'post_date',
				),
			),
			'group_by'     => $this->group_by_query,
			'order_by'     => 'post_date ASC',
			'query_type'   => 'get_results',
			'filter_range' => true,
			'order_types'  => wc_get_order_types( 'sales-reports' ),
			'order_status' => array( 'completed', 'processing', 'on-hold', 'refunded' ),
		) );

		// Add fees in Report Data.
		$this->report_data->stripe_fees = $stripe_fees;

		// Add Total Stripe fees.
		$this->report_data->total_stripe_fees = wc_format_decimal( array_sum( wp_list_pluck( $this->report_data->stripe_fees, 'total' ) ), 2 );

		// Update Net Sale - Stripe fees.
		$this->report_data->net_sales = wc_format_decimal( $this->report_data->total_sales - $this->report_data->total_shipping - max( 0, $this->report_data->total_tax ) - max( 0, $this->report_data->total_shipping_tax ) - max( 0, $this->report_data->total_stripe_fees ), 2 );

		return $this->report_data;
	}

	/**
	 * Unfortunately, the Stripe Gateway extension used meta_key in the past whose name contained a space,
	 * now it uses underscores, but did not put any updater to update the old data.
	 * With this method, the query is modified to be operational.
	 *
	 * @param array $query Request.
	 *
	 * @return array $query
	 */
	public function clean_query_get_order( $query ) {

		preg_match( '/' . self::TWEAK_REQ . '/', $query['select'], $match );
		if ( $match && is_array( $query ) ) {
			$query['select'] = str_replace( self::TWEAK_REQ, 'key', $query['select'] );
			$query['join']   = str_replace( 'meta_' . self::TWEAK_REQ, 'meta_key', $query['join'] );
			$query['join']   = str_replace( "= '" . self::TWEAK_REQ . "'", "IN ('" . self::META_NAME_FEE . "','" . self::LEGACY_META_NAME_FEE . "')", $query['join'] );
		}

		return $query;
	}

	/**
	 * Added query results in the chart.
	 *
	 * @param array $data Data table.
	 *
	 * @return array $this->chart_data
	 */
	public function chart_add_report_data( $data ) {

		// Set $data in class.
		$this->chart_data = $data;

		// Add Stripe fees in data graph array.
		$this->chart_data['stripe_fees'] = $this->prepare_chart_data( $this->report_data->stripe_fees, 'post_date', 'total', $this->chart_interval, $this->start_date, $this->chart_groupby );

		// Update Net Sale - Stripe fees.
		foreach ( $this->chart_data['net_order_amounts'] as $order_amount_key => $order_amount_value ) {
			$this->chart_data['net_order_amounts'][ $order_amount_key ][1] -= $this->chart_data['stripe_fees'][ $order_amount_key ][1];
		}

		return $this->chart_data;
	}

	/**
	 * Add the new legend in chart.
	 */
	public function chart_add_legend() {

		if ( ! $this->chart_data ) {
			return;
		}
		$link = sprintf(
			'<li style="border-color: %s" class="highlight_series" data-series="9">%s</li>',
			self::COLOR,
			sprintf(
			    // translators: %s is Stripe fees.
				_x( '%s Stripe fees', 'number', 'woo-stripe-fee-in-report' ),
				'<strong>' . wc_price( $this->report_data->total_stripe_fees ) . '</strong>'
			)
		);
		?>

        <script type="text/javascript">
            jQuery('ul.chart-legend').append('<?php echo $link; ?>');
        </script>

		<?php

	}

	/**
	 * Replaces hover text for legend.
	 * In our case, I use it to change the text concerning the net commands taking into account the Stripes fees.
	 */
	public function chart_update_legend_placeholder() {
		$update_legend = array(
			7 => esc_html__( 'This is the sum of the order totals after any refunds and excluding shipping, taxes and Stripe fees.', 'woo-stripe-fee-in-report' ), // Net sales amount
		);
		echo '<script type="text/javascript">';
		foreach ( $update_legend as $serie => $text ) {
			printf(
				"jQuery('ul.chart-legend').find('li[data-series=%d]').attr('data-tip', '%s');",
				intval( $serie ),
				esc_js( $text )
			);
		}
		echo '</script>';
	}

	/**
	 * Change the size of the graph to be in harmony with the legend.
	 */
	public function chart_style() {

		echo '<script type="text/javascript">';
		echo 'jQuery(".woocommerce-reports-wide .postbox .chart-placeholder").height(jQuery("ul.chart-legend").height() - 25);';
		echo '</script>';
	}

	/**
	 * There is a way to add data to jQuery.plot, but it is almost useless if you change the graph options.
	 * The ideal would probably be a hook directly in WooCommerce.
	 * Much of the code comes from the WC_Report_Sales_By_Date get_main_chart() method
	 */
	public function chart_draw() {

		global $wp_locale;

		if ( ! $this->chart_data ) {
			return;
		}

		$stripe_fees = wp_json_encode(
			array_map( array( $this, 'round_chart_totals' ), array_values( $this->chart_data['stripe_fees'] ) )
		);

		?>

        <script type="text/javascript">

            jQuery(function () {

                var stripe_main_char;

                var drawGraph = function (highlight) {

                    // Get Data.
                    var series = [];
                    var current_series = main_chart.getData();
                    jQuery(current_series).each(function (i, value) {

                        // Change Yaxis shipping.
                        if (value.label === '<?php echo esc_js( __( 'Shipping amount', 'woocommerce' ) ) ?>') {
                            value.yaxis = 3;
                        }

                        series.push(value);
                    });

                    // Stripe serie.
                    var stripe_series = {
                        label: "<?php echo esc_js( esc_html__( 'Stripe fees', 'woo-stripe-fee-in-report' ) ) ?>",
                        data: <?php echo $stripe_fees;?>,
                        yaxis: 3,
                        color: '<?php echo self::COLOR; ?>',
                        points: {show: true, radius: 5, lineWidth: 2, fillColor: '#fff', fill: true},
                        lines: {show: true, lineWidth: 2, fill: false},
                        shadowSize: 0,
						<?php echo $this->get_currency_tooltip(); ?>
                    };
                    series.push(stripe_series);

                    if (highlight !== 'undefined' && series[highlight]) {
                        highlight_series = series[highlight];

                        highlight_series.color = '#9c5d90';

                        if (highlight_series.bars) {
                            highlight_series.bars.fillColor = '#9c5d90';
                        }

                        if (highlight_series.lines) {
                            highlight_series.lines.lineWidth = 5;
                        }
                    }

                    stripe_main_char = jQuery.plot(
                        jQuery('.chart-placeholder.main'),
                        series,
                        {
                            legend: {
                                show: false
                            },
                            grid: {
                                color: '#aaa',
                                borderColor: 'transparent',
                                borderWidth: 0,
                                hoverable: true
                            },
                            xaxes: [{
                                color: '#aaa',
                                position: "bottom",
                                tickColor: 'transparent',
                                mode: "time",
                                timeformat: "<?php echo ( 'day' === $this->chart_groupby ) ? '%d %b' : '%b'; ?>",
                                monthNames: <?php echo json_encode( array_values( $wp_locale->month_abbrev ) ) ?>,
                                tickLength: 1,
                                minTickSize: [1, "<?php echo $this->chart_groupby; ?>"],
                                font: {
                                    color: "#aaa"
                                }
                            }],
                            yaxes: [
                                {
                                    min: 0,
                                    minTickSize: 1,
                                    tickDecimals: 0,
                                    color: '#d4d9dc',
                                    font: {color: "#aaa"}
                                },
                                {
                                    position: "right",
                                    min: 0,
                                    tickDecimals: 2,
                                    alignTicksWithAxis: 1,
                                    color: 'transparent',
                                    font: {color: "#aaa"}
                                },
                                {
                                    position: "right",
                                    min: 0,
                                    tickDecimals: 2,
                                    alignTicksWithAxis: 1,
                                    color: 'transparent',
                                    font: {color: "#aaa"},
                                    autoscaleMargin: 5
                                }
                            ],
                        }
                    );
                    jQuery('.chart-placeholder').resize();
                }

                drawGraph();

                jQuery('.highlight_series').hover(
                    function () {
                        drawGraph(jQuery(this).data('series'));
                    },
                    function () {
                        drawGraph();
                    }
                );

	            jQuery( '.export_csv' ).click( function() {
		            var exclude_series = jQuery( this ).data( 'exclude_series' ) || '';
		            exclude_series    = exclude_series.toString();
		            exclude_series    = exclude_series.split( ',' );
		            var xaxes_label   = jQuery( this ).data( 'xaxes' );
		            var groupby       = jQuery( this ) .data( 'groupby' );
		            var index_type    = jQuery( this ).data( 'index_type' );
		            var export_format = jQuery( this ).data( 'export' );
		            var csv_data      = 'data:text/csv;charset=utf-8,\uFEFF';
		            var s, series_data, d;

		            if ( 'table' === export_format ) {

			            jQuery( this ).offsetParent().find( 'thead tr,tbody tr' ).each( function() {
				            jQuery( this ).find( 'th, td' ).each( function() {
					            var value = jQuery( this ).text();
					            value = value.replace( '[?]', '' ).replace( '#', '' );
					            csv_data += '"' + value + '"' + ',';
				            });
				            csv_data = csv_data.substring( 0, csv_data.length - 1 );
				            csv_data += '\n';
			            });

			            jQuery( this ).offsetParent().find( 'tfoot tr' ).each( function() {
				            jQuery( this ).find( 'th, td' ).each( function() {
					            var value = jQuery( this ).text();
					            value = value.replace( '[?]', '' ).replace( '#', '' );
					            csv_data += '"' + value + '"' + ',';
					            if ( jQuery( this ).attr( 'colspan' ) > 0 ) {
						            for ( i = 1; i < jQuery(this).attr('colspan'); i++ ) {
							            csv_data += '"",';
						            }
					            }
				            });
				            csv_data = csv_data.substring( 0, csv_data.length - 1 );
				            csv_data += '\n';
			            });

		            } else {

			            if ( ! window.main_chart ) {
				            return false;
			            }

			            var the_series = stripe_main_char.getData();
			            var series     = [];
			            csv_data      += '"' + xaxes_label + '",';

			            jQuery.each( the_series, function( index, value ) {
				            if ( ! exclude_series || jQuery.inArray( index.toString(), exclude_series ) === -1 ) {
					            series.push( value );
				            }
			            });

			            // CSV Headers
			            for ( s = 0; s < series.length; ++s ) {
				            csv_data += '"' + series[s].label + '",';
			            }

			            csv_data = csv_data.substring( 0, csv_data.length - 1 );
			            csv_data += '\n';

			            // Get x axis values
			            var xaxis = {};

			            for ( s = 0; s < series.length; ++s ) {
				            series_data = series[s].data;
				            for ( d = 0; d < series_data.length; ++d ) {
					            xaxis[series_data[d][0]] = [];
					            // Zero values to start
					            for ( var i = 0; i < series.length; ++i ) {
						            xaxis[series_data[d][0]].push(0);
					            }
				            }
			            }

			            // Add chart data
			            for ( s = 0; s < series.length; ++s ) {
				            series_data = series[s].data;
				            for ( d = 0; d < series_data.length; ++d ) {
					            xaxis[series_data[d][0]][s] = series_data[d][1];
				            }
			            }

			            // Loop data and output to csv string
			            jQuery.each( xaxis, function( index, value ) {
				            var date = new Date( parseInt( index, 10 ) );

				            if ( 'none' === index_type ) {
					            csv_data += '"' + index + '",';
				            } else {
					            if ( groupby === 'day' ) {
						            csv_data += '"' + date.getUTCFullYear() + '-' + parseInt( date.getUTCMonth() + 1, 10 ) + '-' + date.getUTCDate() + '",';
					            } else {
						            csv_data += '"' + date.getUTCFullYear() + '-' + parseInt( date.getUTCMonth() + 1, 10 ) + '",';
					            }
				            }

				            for ( var d = 0; d < value.length; ++d ) {
					            var val = value[d];

					            if ( Math.round( val ) !== val ) {
						            val = parseFloat( val );
						            val = val.toFixed( 2 );
					            }

					            csv_data += '"' + val + '",';
				            }
				            csv_data = csv_data.substring( 0, csv_data.length - 1 );
				            csv_data += '\n';
			            } );
		            }

		            // Set data as href and return
		            jQuery( this ).attr( 'href', encodeURI( csv_data ) );
		            return true;
	            });

            });
        </script>
		<?php
	}

	/**
	 * Method from the 'WC_Report_Taxes_By_Date' file required for the construction of the graph
	 *
	 * @param array|string $amount value to transform.
	 *
	 * @return array|string
	 */
	private function round_chart_totals( $amount ) {

		if ( is_array( $amount ) ) {
			return array( $amount[0], wc_format_decimal( $amount[1], wc_get_price_decimals() ) );
		} else {
			return wc_format_decimal( $amount, wc_get_price_decimals() );
		}
	}

}


woo-stripe-fee-report.php
<?php
/**
Plugin Name:		    WooCommerce Stripe fee in Report
Plugin URI:			    https://github.com/rvola/woo-stripe-fee-report

Description:		    Shows Stripe charges in WooCommerce charts. Calculates a correct net total.

Version:			    1.0.0
Revision:			    2018-02-24
Creation:               2018-02-24

Author:				    studio RVOLA
Author URI:			    https://www.rvola.com

Text Domain:		    woo-stripe-fee-report
Domain Path:		    /languages

Requires at least:      4.4
Tested up to:           4.9
Requires PHP:           5.3
WC requires at least:   3.1.0
WC tested up to:        3.3.0

License:                GNU General Public License v3.0
License URI:            https://www.gnu.org/licenses/gpl-3.0.html
**/

namespace RVOLA\WOO\SFR;

if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

include_once ABSPATH . 'wp-admin/includes/plugin.php';
// Check if WooCommerce and WooCommerce Gateway Stripe is loaded.
if (
	is_plugin_active( 'woocommerce/woocommerce.php' )
	&& is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' )
) {
	require_once dirname( __FILE__ ) . '/includes/class-wp.php';
	add_action( 'plugins_loaded', array( 'RVOLA\WOO\SFR\WP', 'instance' ), 10 );
}


woo-stripe-fee-in-report.php
<?php
/**
 * Plugin Name:             WooCommerce Stripe fee in Report
 * Plugin URI:              https://github.com/rvola/woo-stripe-fee-in-report
 *
 * Description:             Shows Stripe fees in WooCommerce charts. Calculates a correct net total.
 *
 * Version:                 1.2.6
 * Revision:                2020-08-24
 * Creation:                2018-02-24
 *
 * Author:                  studio RVOLA
 * Author URI:              https://www.rvola.com
 *
 * Text Domain:             woo-stripe-fee-in-report
 * Domain Path:             /languages
 *
 * Requires at least:       4.4
 * Tested up to:            5.4
 * Requires PHP:            7.0
 * WC requires at least:    3.1
 * WC tested up to:         4.4
 *
 * License:                 GNU General Public License v3.0
 * License URI:             https://www.gnu.org/licenses/gpl-3.0.html
 **/

namespace RVOLA\WOO\SFR;

if ( ! defined( 'ABSPATH' ) ) {
	exit;
}
define( 'WOOSFR_FILE', __FILE__ );

include_once ABSPATH . 'wp-admin/includes/plugin.php';
// Check if WooCommerce and WooCommerce Gateway Stripe is loaded.
if (
	is_plugin_active( 'woocommerce/woocommerce.php' )
	&& is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' )
) {
	require_once dirname( WOOSFR_FILE ) . '/includes/class-wp.php';
	add_action( 'plugins_loaded', array( 'RVOLA\WOO\SFR\WP', 'instance' ), 10 );
}
